## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454312`
Commit: [`e633cb50240739a549e657d6d6471c620ef9e0a0`](https://github.com/NixOS/nixpkgs/commit/e633cb50240739a549e657d6d6471c620ef9e0a0) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/e633cb50240739a549e657d6d6471c620ef9e0a0..pull/454312/head))
Merge: [`9ad56524799ec32b6137913b1aa1f6dd6764df82`](https://github.com/NixOS/nixpkgs/commit/9ad56524799ec32b6137913b1aa1f6dd6764df82)

Logs: https://github.com/qbisi/nixpkgs-review-gha/actions/runs/18730320896


---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.fenics-dolfinx</li>
    <li>python312Packages.fenics-dolfinx.dist</li>
    <li>python313Packages.fenics-dolfinx</li>
    <li>python313Packages.fenics-dolfinx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>dolfinx</li>
    <li>python312Packages.fenics-basix</li>
    <li>python312Packages.fenics-basix.dist</li>
    <li>python312Packages.fenics-ffcx</li>
    <li>python312Packages.fenics-ffcx.dist</li>
    <li>python312Packages.slepc4py</li>
    <li>python313Packages.fenics-basix</li>
    <li>python313Packages.fenics-basix.dist</li>
    <li>python313Packages.fenics-ffcx</li>
    <li>python313Packages.fenics-ffcx.dist</li>
    <li>python313Packages.slepc4py</li>
    <li>slepc</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.fenics-dolfinx</summary>
<pre>unit/refinement/test_uniform.py ......                                   [100%]

=================================== FAILURES ===================================
_________________________________ test_version _________________________________

    def test_version():
        &quot;&quot;&quot;Test that installed Python version matches C++ version.&quot;&quot;&quot;
        # Change any final &#x27;.dev0&#x27; to &#x27;.0&#x27;
        py_version = dolfinx.__version__.replace(&quot;dev0&quot;, &quot;0&quot;)
        cpp_version = dolfinx.cpp.__version__
        if py_version != cpp_version:
&gt;           raise RuntimeError(
                f&quot;Incorrect versions. Python version: {py_version}, Core version: {cpp_version}&quot;
            )
E           RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core version: 0.10.0

unit/common/test_version.py:17: RuntimeError
=========================== short test summary info ============================
FAILED unit/common/test_version.py::test_version - RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core versio...
===== 1 failed, 2863 passed, 87 skipped, 27 xfailed in 1107.93s (0:18:27) ======</pre>
</details>
<details>
<summary>python313Packages.fenics-dolfinx</summary>
<pre>unit/refinement/test_uniform.py ......                                   [100%]

=================================== FAILURES ===================================
_________________________________ test_version _________________________________

    def test_version():
        &quot;&quot;&quot;Test that installed Python version matches C++ version.&quot;&quot;&quot;
        # Change any final &#x27;.dev0&#x27; to &#x27;.0&#x27;
        py_version = dolfinx.__version__.replace(&quot;dev0&quot;, &quot;0&quot;)
        cpp_version = dolfinx.cpp.__version__
        if py_version != cpp_version:
&gt;           raise RuntimeError(
                f&quot;Incorrect versions. Python version: {py_version}, Core version: {cpp_version}&quot;
            )
E           RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core version: 0.10.0

unit/common/test_version.py:17: RuntimeError
=========================== short test summary info ============================
FAILED unit/common/test_version.py::test_version - RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core versio...
===== 1 failed, 2863 passed, 87 skipped, 27 xfailed in 1082.34s (0:18:02) ======</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.fenics-dolfinx</li>
    <li>python312Packages.fenics-dolfinx.dist</li>
    <li>python313Packages.fenics-dolfinx</li>
    <li>python313Packages.fenics-dolfinx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>dolfinx</li>
    <li>python312Packages.fenics-basix</li>
    <li>python312Packages.fenics-basix.dist</li>
    <li>python312Packages.fenics-ffcx</li>
    <li>python312Packages.fenics-ffcx.dist</li>
    <li>python312Packages.slepc4py</li>
    <li>python313Packages.fenics-basix</li>
    <li>python313Packages.fenics-basix.dist</li>
    <li>python313Packages.fenics-ffcx</li>
    <li>python313Packages.fenics-ffcx.dist</li>
    <li>python313Packages.slepc4py</li>
    <li>slepc</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.fenics-dolfinx</summary>
<pre>unit/refinement/test_uniform.py ......                                   [100%]

=================================== FAILURES ===================================
_________________________________ test_version _________________________________

    def test_version():
        &quot;&quot;&quot;Test that installed Python version matches C++ version.&quot;&quot;&quot;
        # Change any final &#x27;.dev0&#x27; to &#x27;.0&#x27;
        py_version = dolfinx.__version__.replace(&quot;dev0&quot;, &quot;0&quot;)
        cpp_version = dolfinx.cpp.__version__
        if py_version != cpp_version:
&gt;           raise RuntimeError(
                f&quot;Incorrect versions. Python version: {py_version}, Core version: {cpp_version}&quot;
            )
E           RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core version: 0.10.0

unit/common/test_version.py:17: RuntimeError
=========================== short test summary info ============================
FAILED unit/common/test_version.py::test_version - RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core versio...
====== 1 failed, 2863 passed, 87 skipped, 27 xfailed in 864.86s (0:14:24) ======</pre>
</details>
<details>
<summary>python313Packages.fenics-dolfinx</summary>
<pre>unit/refinement/test_uniform.py ......                                   [100%]

=================================== FAILURES ===================================
_________________________________ test_version _________________________________

    def test_version():
        &quot;&quot;&quot;Test that installed Python version matches C++ version.&quot;&quot;&quot;
        # Change any final &#x27;.dev0&#x27; to &#x27;.0&#x27;
        py_version = dolfinx.__version__.replace(&quot;dev0&quot;, &quot;0&quot;)
        cpp_version = dolfinx.cpp.__version__
        if py_version != cpp_version:
&gt;           raise RuntimeError(
                f&quot;Incorrect versions. Python version: {py_version}, Core version: {cpp_version}&quot;
            )
E           RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core version: 0.10.0

unit/common/test_version.py:17: RuntimeError
=========================== short test summary info ============================
FAILED unit/common/test_version.py::test_version - RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core versio...
====== 1 failed, 2863 passed, 87 skipped, 27 xfailed in 861.78s (0:14:21) ======</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.fenics-basix</li>
    <li>python312Packages.fenics-basix.dist</li>
    <li>python312Packages.fenics-ffcx</li>
    <li>python312Packages.fenics-ffcx.dist</li>
    <li>python313Packages.fenics-basix</li>
    <li>python313Packages.fenics-basix.dist</li>
    <li>python313Packages.fenics-ffcx</li>
    <li>python313Packages.fenics-ffcx.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.fenics-dolfinx</li>
    <li>python312Packages.fenics-dolfinx.dist</li>
    <li>python313Packages.fenics-dolfinx</li>
    <li>python313Packages.fenics-dolfinx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>dolfinx</li>
    <li>python312Packages.fenics-basix</li>
    <li>python312Packages.fenics-basix.dist</li>
    <li>python312Packages.fenics-ffcx</li>
    <li>python312Packages.fenics-ffcx.dist</li>
    <li>python312Packages.slepc4py</li>
    <li>python313Packages.fenics-basix</li>
    <li>python313Packages.fenics-basix.dist</li>
    <li>python313Packages.fenics-ffcx</li>
    <li>python313Packages.fenics-ffcx.dist</li>
    <li>python313Packages.slepc4py</li>
    <li>slepc</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.fenics-dolfinx</summary>
<pre>  Module &#x27;adios2&#x27; was found, but when imported by pytest it raised:
      ImportError(&quot;dlopen(/nix/store/daanhwp9vqsxqbpi4pcs75g0n99x0a34-adios2-2.10.2/lib/python3.12/site-packages/adios2/bindings/adios2_bindings.cpython-312-darwin.so, 0x0002): Library not loaded: /private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib\n  Referenced from: &lt;CA4033D1-93D8-322C-8054-6BFD370FF933&gt; /nix/store/daanhwp9vqsxqbpi4pcs75g0n99x0a34-adios2-2.10.2/lib/python3.12/site-packages/adios2/bindings/adios2_bindings.cpython-312-darwin.so\n  Reason: tried: &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/System/Volumes/Preboot/Cryptexes/OS/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/local/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file, not in dyld cache)&quot;)
  In pytest 9.1 this warning will become an error by default.
  You can fix the underlying problem, or alternatively overwrite this behavior and silence this warning by passing exc_type=ImportError explicitly.
  See https://docs.pytest.org/en/stable/deprecations.html#pytest-importorskip-default-behavior-regarding-importerror
    adios2 = pytest.importorskip(&quot;adios2&quot;, minversion=&quot;2.10.0&quot;)

test/unit/io/test_adios2.py::TestVTX::test_dg_0_data
  /nix/build/nix-6116-404977198/source/python/test/unit/io/test_adios2.py:232: PytestDeprecationWarning: 
  Module &#x27;adios2&#x27; was found, but when imported by pytest it raised:
      ImportError(&quot;dlopen(/nix/store/daanhwp9vqsxqbpi4pcs75g0n99x0a34-adios2-2.10.2/lib/python3.12/site-packages/adios2/bindings/adios2_bindings.cpython-312-darwin.so, 0x0002): Library not loaded: /private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib\n  Referenced from: &lt;CA4033D1-93D8-322C-8054-6BFD370FF933&gt; /nix/store/daanhwp9vqsxqbpi4pcs75g0n99x0a34-adios2-2.10.2/lib/python3.12/site-packages/adios2/bindings/adios2_bindings.cpython-312-darwin.so\n  Reason: tried: &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/System/Volumes/Preboot/Cryptexes/OS/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/local/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file, not in dyld cache)&quot;)
  In pytest 9.1 this warning will become an error by default.
  You can fix the underlying problem, or alternatively overwrite this behavior and silence this warning by passing exc_type=ImportError explicitly.
  See https://docs.pytest.org/en/stable/deprecations.html#pytest-importorskip-default-behavior-regarding-importerror
    adios2 = pytest.importorskip(&quot;adios2&quot;, minversion=&quot;2.10.0&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED unit/common/test_version.py::test_version - RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core versio...
= 1 failed, 2854 passed, 96 skipped, 27 xfailed, 9 warnings in 1049.90s (0:17:29) =</pre>
</details>
<details>
<summary>python313Packages.fenics-dolfinx</summary>
<pre>  Module &#x27;adios2&#x27; was found, but when imported by pytest it raised:
      ImportError(&quot;dlopen(/nix/store/6023x7g0kj60z13pnqm1whmyvhscjilz-adios2-2.10.2/lib/python3.13/site-packages/adios2/bindings/adios2_bindings.cpython-313-darwin.so, 0x0002): Library not loaded: /private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib\n  Referenced from: &lt;B5E662E1-2DA8-344F-AE5A-9FC8B2177B14&gt; /nix/store/6023x7g0kj60z13pnqm1whmyvhscjilz-adios2-2.10.2/lib/python3.13/site-packages/adios2/bindings/adios2_bindings.cpython-313-darwin.so\n  Reason: tried: &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/System/Volumes/Preboot/Cryptexes/OS/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/local/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file, not in dyld cache)&quot;)
  In pytest 9.1 this warning will become an error by default.
  You can fix the underlying problem, or alternatively overwrite this behavior and silence this warning by passing exc_type=ImportError explicitly.
  See https://docs.pytest.org/en/stable/deprecations.html#pytest-importorskip-default-behavior-regarding-importerror
    adios2 = pytest.importorskip(&quot;adios2&quot;, minversion=&quot;2.10.0&quot;)

test/unit/io/test_adios2.py::TestVTX::test_dg_0_data
  /nix/build/nix-6116-404977197/source/python/test/unit/io/test_adios2.py:232: PytestDeprecationWarning: 
  Module &#x27;adios2&#x27; was found, but when imported by pytest it raised:
      ImportError(&quot;dlopen(/nix/store/6023x7g0kj60z13pnqm1whmyvhscjilz-adios2-2.10.2/lib/python3.13/site-packages/adios2/bindings/adios2_bindings.cpython-313-darwin.so, 0x0002): Library not loaded: /private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib\n  Referenced from: &lt;B5E662E1-2DA8-344F-AE5A-9FC8B2177B14&gt; /nix/store/6023x7g0kj60z13pnqm1whmyvhscjilz-adios2-2.10.2/lib/python3.13/site-packages/adios2/bindings/adios2_bindings.cpython-313-darwin.so\n  Reason: tried: &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/System/Volumes/Preboot/Cryptexes/OS/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/private/tmp/nix-build-adios2-2.10.2.drv-0/source/build/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/local/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file), &#x27;/usr/lib/libadios2_cxx11_mpi.2.10.dylib&#x27; (no such file, not in dyld cache)&quot;)
  In pytest 9.1 this warning will become an error by default.
  You can fix the underlying problem, or alternatively overwrite this behavior and silence this warning by passing exc_type=ImportError explicitly.
  See https://docs.pytest.org/en/stable/deprecations.html#pytest-importorskip-default-behavior-regarding-importerror
    adios2 = pytest.importorskip(&quot;adios2&quot;, minversion=&quot;2.10.0&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED unit/common/test_version.py::test_version - RuntimeError: Incorrect versions. Python version: 0.10.0.post0, Core versio...
= 1 failed, 2854 passed, 96 skipped, 27 xfailed, 9 warnings in 1059.69s (0:17:39) =</pre>
</details>
</details>

--- a/pkgs/by-name/do/dolfinx/package.nix
+++ b/pkgs/by-name/do/dolfinx/package.nix
@@ -12,6 +12,7 @@
   kahip,
   adios2,
   python3Packages,
+  darwinMinVersionHook,
   catch2_3,
   withParmetis ? false,
 }:
@@ -25,14 +26,14 @@ let
   );
 in
 stdenv.mkDerivation (finalAttrs: {
-  version = "0.9.0.post1";
+  version = "0.10.0.post1";
   pname = "dolfinx";
 
   src = fetchFromGitHub {
     owner = "fenics";
     repo = "dolfinx";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-4IIx7vUZeDwOGVdyC2PBvfhVjrmGZeVQKAwgDYScbY0=";
+    hash = "sha256-ZsaEcJdvsf3dxJ739/CU20+drjbAvuc/HkIGCfh9U5A=";
   };
 
   preConfigure = ''
@@ -48,6 +49,7 @@ stdenv.mkDerivation (finalAttrs: {
     dolfinxPackages.kahip
     dolfinxPackages.scotch
   ]
+  ++ lib.optional stdenv.hostPlatform.isDarwin (darwinMinVersionHook "13.3")
   ++ lib.optional withParmetis dolfinxPackages.parmetis;
 
   propagatedBuildInputs = [

--- a/pkgs/by-name/sl/slepc/package.nix
+++ b/pkgs/by-name/sl/slepc/package.nix
@@ -86,7 +86,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   pythonImportsCheck = [ "slepc4py" ];
 
-  shellHook = ./setup-hook.sh;
+  setupHook = ./setup-hook.sh;
 
   meta = {
     description = "Scalable Library for Eigenvalue Problem Computations";

--- a/pkgs/development/python-modules/fenics-dolfinx/default.nix
+++ b/pkgs/development/python-modules/fenics-dolfinx/default.nix
@@ -16,6 +16,7 @@
 
   # buildInputs
   dolfinx,
+  darwinMinVersionHook,
 
   # dependency
   numpy,
@@ -63,7 +64,6 @@ buildPythonPackage rec {
   pyproject = true;
 
   pythonRelaxDeps = [
-    "cffi"
     "fenics-ufl"
   ];
 
@@ -87,7 +87,8 @@ buildPythonPackage rec {
 
   buildInputs = [
     fenicsPackages.dolfinx
-  ];
+  ]
+  ++ lib.optional stdenv.hostPlatform.isDarwin (darwinMinVersionHook "13.3");
 
   dependencies = [
     numpy
@@ -102,8 +103,6 @@ buildPythonPackage rec {
     (mpi4py.override { inherit (fenicsPackages) mpi; })
   ];
 
-  doCheck = true;
-
   nativeCheckInputs = [
     scipy
     matplotlib
@@ -113,19 +112,11 @@ buildPythonPackage rec {
   ];
 
   preCheck = ''
-    rm -rf dolfinx
+    cd test
   '';
 
   pythonImportsCheck = [
     "dolfinx"
-  ];
-
-  disabledTests = [
-    # require cffi<1.17
-    "test_cffi_expression"
-    "test_hexahedron_mesh"
-    # https://github.com/FEniCS/dolfinx/issues/1104
-    "test_cube_distance"
   ];
 
   passthru = {

--- a/pkgs/development/python-modules/fenics-ffcx/default.nix
+++ b/pkgs/development/python-modules/fenics-ffcx/default.nix
@@ -1,6 +1,5 @@
 {
   lib,
-  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
   setuptools,
@@ -16,14 +15,14 @@
 
 buildPythonPackage rec {
   pname = "fenics-ffcx";
-  version = "0.9.0";
+  version = "0.10.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "fenics";
     repo = "ffcx";
     tag = "v${version}";
-    hash = "sha256-eAV//RLbrxyhqgbZ2DiR7qML7xfgPn0/Seh+2no0x8w=";
+    hash = "sha256-i8fawnXWxIHfOvb0nK4/bzhrzfRJJACMCkFZKtdUwkU=";
   };
 
   pythonRelaxDeps = [
@@ -52,8 +51,6 @@ buildPythonPackage rec {
     pytestCheckHook
     addBinToPathHook
   ];
-
-  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=unused-command-line-argument";
 
   meta = {
     homepage = "https://fenicsproject.org";

